prereise/gather/demanddata/transportation_electrification/immediate.py

```diff
@@ -0,0 +1,298 @@
+
+    TRANS_charge = np.zeros(24 * len(input_day))
+    data_day = mileage.get_data_day(newdata)
+
+    trip_strategy = 1
+    kwh = kwhmi * veh_range
+    weekday_electricload, weekend_electricload = np.zeros(4800), np.zeros(4800)
+
+    l = len(newdata)
+
+    # weekday first
+    trip_num = 1
+    start_point = 0
+    end_point = 0
+
+    for i in range(l):
+        if data_day[i] == 2:  # 2 is weekday
+            if (
+                i > 0
+                and newdata.iloc[i, newdata.columns.get_loc("sample vehicle number")]
+                == newdata.iloc[i - 1, newdata.columns.get_loc("sample vehicle number")]
+            ):
+                newdata.iloc[
+                    i, newdata.columns.get_loc("trip start battery charge")
+                ] = (
+                    newdata.iloc[
+                        i - 1, newdata.columns.get_loc("trip end battery charge")
+                    ]
+                    + newdata.iloc[
+                        i - 1, newdata.columns.get_loc("charging consumption")
+                    ]
+                )
+                trip_num += 1
+                newdata.iloc[i, newdata.columns.get_loc("trip number")] = trip_num
+
+            else:
+                newdata.iloc[
+                    i, newdata.columns.get_loc("trip start battery charge")
+                ] = kwh
+                trip_num = 1
+                newdata.iloc[i, newdata.columns.get_loc("trip number")] = trip_num
+
+            if (
+                newdata.iloc[i, newdata.columns.get_loc("total vehicle miles traveled")]
+                < veh_range * const.ER
+            ):
+                newdata.iloc[i, newdata.columns.get_loc("BEV could be used")] = 1
+                newdata.iloc[i, newdata.columns.get_loc("trip end battery charge")] = (
+                    newdata.iloc[
+                        i, newdata.columns.get_loc("trip start battery charge")
+                    ]
+                    - newdata.iloc[i, newdata.columns.get_loc("Miles traveled")]
+                    * kwhmi
+                    * const.ER
+                )
+            else:
+                newdata.iloc[i, newdata.columns.get_loc("BEV could be used")] = 0
+
+            trip_data = newdata.iloc[i]
+
+            newdata.iloc[
+                i, newdata.columns.get_loc("charging power")
+            ] = charging.get_charging_power(
+                power,
+                trip_strategy,
+                location_strategy,
+                kwh,
+                trip_data,
+            )
+
+            newdata.iloc[
+                i, newdata.columns.get_loc("charging time")
+            ] = charging.get_charging_time(
+                newdata.iloc[i, newdata.columns.get_loc("charging power")],
+                kwh,
+                newdata.iloc[i, newdata.columns.get_loc("trip end battery charge")],
+                const.charging_efficiency,
+                trip_data,
+            )
```

Comment:
These functions could also be used outside the loop via an `apply` call which applies a function to each row/column of a dataframe. E.g.:
```python
newdata["charging power"] = newdata.apply(
    lambda x: charging.get_charging_power(
        power,
        trip_strategy,
        location_strategy,
        kwh,
        trip_data=x,
    ),
    axis=1,
```
The `lambda` function allows concise creation of one-time functions, in this case the function takes a single input (`x`) and returns the output of `charging.get_charging_power` called with `x` for `trip_data`, plus the other parameters. `axis=1` means that the function is called on each row of data, rather than each column.